lib/remote/zone.cpp

```diff
@@ -125,10 +125,13 @@ bool Zone::IsGlobal() const
 	return GetGlobal();
 }
 
-bool Zone::IsSingleInstance() const
+bool Zone::IsHACluster() const
 {
-	Array::Ptr endpoints = GetEndpointsRaw();
-	return !endpoints || endpoints->GetLength() < 2;
+	if (auto endpoints(GetEndpointsRaw()); endpoints) {
+		return endpoints->GetLength() >= 2;
+	}
+
+	return false;
```

Comment:
This, in contrast, looks unnecessarily inflated. (Both diff and resulting code.)

```suggestion
	Array::Ptr endpoints = GetEndpointsRaw();
	return endpoints && endpoints->GetLength() >= 2;
```